lib/buildsystems/ruby.rb

```diff
@@ -1,22 +1,58 @@
-require_relative '../const'
-require_relative '../package'
+require 'active_support/core_ext/object/blank'
+require 'color'
+require 'package'
+
+def check_gem_binary_build_needed(gem_name = nil, gem_ver = nil)
+  puts "Checking to see if gem compile for #{gem_name} is needed..."
+  @extract_dir = "#{name}.#{Time.now.utc.strftime('%Y%m%d%H%M%S')}.dir"
+  FileUtils.mkdir_p File.join(CREW_BREW_DIR, @extract_dir)
+  Dir.chdir(File.join(CREW_BREW_DIR, @extract_dir)) do
+    # Need to check if the gem has extensions. If it does, we need
+    # either a compiler or a pre-compiled binary gem.
+    system "gem fetch #{gem_name} --platform=ruby --version=#{gem_ver}"
+    system "gem unpack #{gem_name}-#{gem_ver}.gem"
+    return system "grep -q -r spec.extensions  #{gem_name}-#{gem_ver}/*.gemspec", %i[out err] => File::NULL
+  end
+end
+
+def set_vars(passed_name = nil, passed_version = nil)
+  # crewlog "Setting gem variables... name: #{passed_name}, version: #{passed_version}"
+  # This assumes the package class name starts with 'Ruby_' and
+  # version is in the form '(gem version)-ruby-(ruby version)'.
+  # For example, name 'Ruby_awesome' and version '1.0.0-ruby-3.3'.
+  @gem_name = passed_name.sub('ruby_', '').sub('_', '-')
+  @gem_ver = passed_version.split('-').first.to_s
+end
 
 class RUBY < Package
-  property :ruby_install_extras
+  property :gem_name, :gem_ver, :ruby_install_extras
 
   depends_on 'ruby'
 
-  no_compile_needed
+  def self.preflight
+    set_vars(name, version)
+    crewlog "@gem_name: #{@gem_name}, @gem_ver: #{@gem_ver}"
+  end
+
+  def self.preinstall
+    @gem_binary_build_needed = check_gem_binary_build_needed(@gem_name, @gem_version) unless no_compile_needed? || gem_compile_needed?
```

Comment:
`unless no_compile_needed? || !gem_compile_needed?`

Reply:
The logic is to not run the check if we already know (as specified in the package file) that we need to (or don't need to) build the binary gem.

Reply:
So if `gem_compile_needed` is set, no `@gem_binary_build_needed`?

Reply:
Right. The latter is a variable set when we don't know whether the gem needs binaries yet.